tests/rest/client/v1/test_login.py

```diff
@@ -748,3 +749,74 @@ def test_login_jwt_invalid_signature(self):
             channel.json_body["error"],
             "JWT validation failed: Signature verification failed",
         )
+
+
+AS_USER = "as_user_alice"
+
+
+class AppserviceLoginRestServletTestCase(unittest.HomeserverTestCase):
+    servlets = [
+        login.register_servlets,
+        register.register_servlets,
+        lambda hs, http_server: WhoamiRestServlet(hs).register(http_server),
```

Comment:
Looks like this gets registered with the rest of the account ones, did you do the lambda just to avoid registering all of them?

I think the code looks fine, I'd like to see some tests for the error scenarioes. The main one I can think of is:

* Trying to login as a user that the appservice isn't interested in.

I suspect the other cases are all pretty covered already?

Reply:
>Looks like this gets registered with the rest of the account ones, did you do the lambda just to avoid registering all of them?

Heh, no I copy and pasted it without thinking. Fixed.

> I'd like to see some tests for the error scenarios.

Added tests for all the other scenarios I could think of.